executor/executor.go

```diff
@@ -72,8 +71,8 @@ func NewExecutor(
 		scanner: scanner,
 		dryRun:  dryRun,
 
-		workersCount: workersCount,
-
+		workersCount:  workersCount,
+		inProgressJobs:   map[string]bool{},
```

Comment:
You don't need it to be map[string]bool, You don't use the boolean we can just add and remove from array "if exist" then ignore else add to the array

Reply:
if exist is o(n) but the map check is o(1)

Reply:
yes but in this case, the number of inProgress jobs expected to be 1 and this to prevent if the same job from being executed while it has been already executing so in every scenario it will be only one job in the array and for readability and reduce the complexity we don't need map array is enough

Reply:
The number of in progress jobs may increase in the future so this should be kept generic. I think a map is more suitable for this implementation. You can use a `map[string]struct{}` if you want to save memory as size of `struct{}` is zero, but that won't be necessary.

Reply:
if you looked to the current flow you will find that we can have multiple decisions at the same time because the query in automation service can take another job and not the same job every time and as islam said we can have multiple ones in the future